examples/vision/text_image_denoiser.py

```diff
@@ -0,0 +1,295 @@
+
+valid_set = tf.data.Dataset.from_tensor_slices((val_source, val_targets))
+valid_set = valid_set.map(data_preprocess, tf.data.AUTOTUNE).batch(BATCH_SIZE)
+
+"""
+## Dataset Visualiztion
+"""
+
+fig, ax = plt.subplots(1, 2, figsize=(16, 8))
+ax = ax.flatten()
+for i, f in enumerate(valid_set.take(1)):
+    noise, clean = f
+    noise = denormalize(noise[i].numpy())
+    clean = denormalize(clean[i].numpy())
+    ax[0].imshow(noise.astype(np.int))
+    ax[0].set_xlabel("source")
+    ax[1].imshow(clean.astype(np.int))
+    ax[1].set_xlabel("target")
+fig.text(x=0.45, y=0.78, s="Dataset Visualization", fontweight="bold")
+fig.set_tight_layout(None)
+
+"""
+## Model
+
+We have used Pre-trained ```ResNet50V2``` U-Net as Encoder, The Feature Maps are
+extracted at different levels forming Downscaling path, Then ```Sub-Pixel Layers``` on
+the Upscaling path with skip-connections.
+"""
+
+
+def conv_unit(x, filters=64, padding="same"):
+    x = keras.layers.Conv2D(filters=filters, kernel_size=3, padding=padding)(x)
+    x = keras.layers.BatchNormalization()(x)
+    x = keras.layers.Activation("relu")(x)
+    return x
+
+
+def subpixel_upsample(x, target_channels, upscale_ratio):
+    estimated_filters = target_channels * (upscale_ratio ** 2)
+    x = keras.layers.Conv2D(estimated_filters, 3, padding="same")(x)
+    x = keras.layers.BatchNormalization()(x)
+    x = keras.layers.LeakyReLU(0.2)(x)
+    x = tf.nn.depth_to_space(x, block_size=upscale_ratio)
+    return x
+
+
+def upscale_unit(previous_input, current_input, target_channels=3, upscale_ratio=2):
+    previous_low_scale = subpixel_upsample(
+        previous_input, target_channels=target_channels, upscale_ratio=upscale_ratio
+    )
+    current_high_scale = conv_unit(current_input, filters=target_channels)
+    concat_scales = tf.concat((previous_low_scale, current_high_scale), axis=-1)
+    feature_manifold = conv_unit(concat_scales, filters=target_channels)
+    return feature_manifold
+
+
+def denoiser(height, width):
+    base_model = resnet_v2.ResNet50V2(
+        include_top=False, weights="imagenet", input_shape=(height, width, 3)
+    )
+    encode0 = base_model.get_layer("input_1").output  # (None,352,608,3)
+    encode1 = base_model.get_layer("conv1_conv").output  # (None,176,304,64)
+    encode2 = base_model.get_layer("conv2_block3_1_relu").output  # (None,88,152,64)
+    encode3 = base_model.get_layer("conv3_block4_1_relu").output  # (None,44,76,128)
+
+    decode1 = upscale_unit(
+        previous_input=encode3, current_input=encode2, target_channels=128
+    )
+    decode2 = upscale_unit(
+        previous_input=decode1, current_input=encode1, target_channels=64
+    )
+    decode3 = upscale_unit(
+        previous_input=decode2, current_input=encode0, target_channels=32
+    )
+
+    out_conv = keras.layers.Conv2D(3, 5, padding="same", activation="sigmoid")(decode3)
+    denoiser_model = keras.Model(base_model.input, out_conv, name="denoiser")
+    #     denoiser_model.summary()
```

Comment:
@anish9 

I think you comment on this because when you call the model, it prints the model summary. In that case, use model summary or `plot_model` outside the function scope.